packages/flame/lib/src/nine_tile_box.dart

```diff
@@ -41,20 +44,114 @@ class NineTileBox {
       : tileSize = tileSize ?? sprite.src.width ~/ 3 {
     this.destTileSize = destTileSize ?? this.tileSize;
     final centerEdge = this.tileSize.toDouble();
-    _center = Rect.fromLTWH(centerEdge, centerEdge, centerEdge, centerEdge);
+    center = Rect.fromLTWH(centerEdge, centerEdge, centerEdge, centerEdge);
     _dst = Rect.fromLTWH(0, 0, this.destTileSize * 3, this.destTileSize * 3);
   }
 
+  /// Creates a nine-box instance with the specified grid size
+  ///
+  /// The outside columns and rows can be specified with fixed width or height.
+  /// The center column and row take up the remaining space.
+  NineTileBox.withGrid(
+    this.sprite, {
+    int leftColumnWidth = 0,
+    int rightColumnWidth = 0,
+    int topRowHeight = 0,
+    int bottomRowHeight = 0,
```

Comment:
It probably makes sense to have these as `double` since that is what is passed in to the `Rect`?

Reply:
Not sure about that one - as far as the user is concerned, they should be integer (partial pixels are not supported). It's unlucky that the Rect needs them as double. 

I can change them to double, but then we'd have to round (or floor) them in case the user passes in partial values?

Reply:
What would happen if they passed in non integer values and we didn't round them? Wouldn't it still work?

Reply:
Yup, that works great. Looks like `Canvas.drawImageNine` uses `round` internally (not floor).
Made the changes.